WordPress/src/main/java/org/wordpress/android/ui/posts/PostsListActivity.kt

```diff
@@ -276,6 +287,11 @@ class PostsListActivity : AppCompatActivity(),
         ActivityId.trackLastActivity(ActivityId.POSTS)
     }
 
+    override fun onPause() {
+        super.onPause()
+        progressDialog?.dismiss()
```

Comment:
The dialog is re-shown when the activity gets re-created, but I think it'll simply disappear if the activity is just paused and resumed. Wdyt?

Note: I'm not 100% sure about this, but I didn't have time to give it a try. I thought it'd be better mentioning it.

Reply:
Good catch by just having a look at the code 😅! I regret to admit this was an oversight from my side! It was part of a snippet I missed to remove while exploring a path for dialog dismiss; finally just added this general precaution adding a preventive dismiss here in ProgressDialogHelper.kt

``` kt
             is ProgressDialogUiState.VisibleProgressDialog -> {
                dialog?.dismiss()
```

(maybe redundant but not harmful and localized) and stayed with what was already in the EditPostActivity (or at least that was the intention 🙂) keeping similar also the PostsListActivity dialog management. Blame on me and really thanks for having pointed it out 🙇‍♂️. 

While reviewing this point I realized that in PostsListActivity I did not managed the Progress Dialog state in cases when the activity is destroied while in background (same as when Don't keep activities developer option is on), so I added this management that is already present in EditPostActivity.

Changes here f93f82ee76a1cbdec795279dd93150cf39316c6f.

Reply:
>I regret to admit this was an oversight from my side! It was part of a snippet I missed to remove while exploring a path for dialog dismiss; 

No worries at all, dialogs can be pretty tricky. However, I think the "fix" has broken something else. If we don't dismiss the dialog when the activity is about to be destroyed, I believe it'll leak the view. 

>While reviewing this point I realized that in PostsListActivity I did not managed the Progress Dialog state in cases when the activity is destroied while in background (same as when Don't keep activities developer option is on), so I added this management that is already present in EditPostActivity.

Hmm I'm wondering if this is what we want. The ViewModel will survive (and retain the state) in most cases and if it doesn't survive its highly probable that the app was in the background for a longer period of time. I think in an ideal scenario we'd cancel the request when the ViewModel's "onCleared" method was called (however, I'm a bit afraid it's not a simple task with the current implementation of PostStore).

I also noticed the dialog is not cancelable - it seems a bit inconvenient for the user. Imagine they unintentionally click on "Preview" and they are on a poor connection - they end up with a blocking dialog for 30+ seconds. 

I think these two issues might actually result in an infinite dialog in some cases. I'm not sure if it can really happen, but my gut tells me it's risky. Imagine both the activity and the VM is destroyed -> we unsubscribe from the dispatcher in `PostListEventListener.onDestroy` -> we miss the `OnPostChanged` event -> the user re-opens the app -> we restore the progress dialog's state -> we will wait indefinitely for an event which will never arrive (since it already arrive)..... This is an issue which you could have not seen coming, so even if it can happen don't worry about it. 

Both my concerns are related to canceling the ongoing request. I can't think of an easy and clean solution right now. I also don't want to make this PR more complicated. I'd consider creating an issue and fixing it later when we have some time to think about it. 
Wdyt? cc @maxme

Reply:
Created ticket #10141 to trace this and update list of open items at the end of this PR description.